cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/detector/notifier/SelfHealingNotifier.java

```diff
@@ -89,7 +89,7 @@ public AnomalyNotificationResult onMetricAnomaly(KafkaMetricAnomaly metricAnomal
   }
 
   @Override
-  public boolean setSelfHealingFor(AnomalyType anomalyType, boolean isSelfHealingEnabled) {
+  public synchronized boolean setSelfHealingFor(AnomalyType anomalyType, boolean isSelfHealingEnabled) {
```

Comment:
Should we move the `synchronized` to `KafkaCruiseControl#handleAdminRequest(AdminParameters)`?